src/main/java/jimmy/mcgymmy/logic/commands/FindCommand.java

```diff
@@ -17,22 +22,78 @@
+    );
+    private OptionalParameter<TagContainsKeywordsPredicate> tagPredicateParameter = this.addOptionalParameter(
+            "tag",
+            "t",
+            "Tag associated with the Food",
+            "Lunch", (s) -> new TagContainsKeywordsPredicate(Arrays.asList(s.split("\\s+")))
+    );
+    private OptionalParameter<DatePredicate> datePredicateParameter = this.addOptionalParameter(
+            "date",
+            "d",
+            "Date associated with the Food",
+            "20-04-2020", (s) -> new DatePredicate(s)
+    );
 
-    void setParameters(Parameter<FoodContainsKeywordsPredicate> predicateParameter) {
-        this.predicateParameter = predicateParameter;
+    void setParameters(OptionalParameter<FoodContainsKeywordsPredicate> foodPredicateParameter,
+                       OptionalParameter<NameContainsKeywordsPredicate> namePredicateParameter,
+                       OptionalParameter<TagContainsKeywordsPredicate> tagPredicateParameter,
+                       OptionalParameter<DatePredicate> datePredicateParameter) {
+        this.foodPredicateParameter = foodPredicateParameter;
+        this.namePredicateParameter = namePredicateParameter;
+        this.tagPredicateParameter = tagPredicateParameter;
+        this.datePredicateParameter = datePredicateParameter;
     }
 
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
-        FoodContainsKeywordsPredicate predicate = predicateParameter.consume();
-        model.updateFilteredFoodList(predicate);
+        FoodContainsKeywordsPredicate foodPredicate = foodPredicateParameter.getValue().orElse(null);
+        NameContainsKeywordsPredicate namePredicate = namePredicateParameter.getValue().orElse(null);
+        TagContainsKeywordsPredicate tagPredicate = tagPredicateParameter.getValue().orElse(null);
+        DatePredicate datePredicate = datePredicateParameter.getValue().orElse(null);
+
+        Predicate<Food> combinedPredicate = null;
+        if (foodPredicate != null) {
+            combinedPredicate = foodPredicate;
+        }
+        if (namePredicate != null) {
+            if (combinedPredicate != null) {
+                combinedPredicate = combinedPredicate.and(namePredicate);
+            } else {
+                combinedPredicate = namePredicate;
+            }
+        }
+        if (tagPredicate != null) {
+            if (combinedPredicate != null) {
+                combinedPredicate = combinedPredicate.and(tagPredicate);
+            } else {
+                combinedPredicate = tagPredicate;
+            }
+        }
+        if (datePredicate != null) {
+            if (combinedPredicate != null) {
+                combinedPredicate = combinedPredicate.and(datePredicate);
+            } else {
+                combinedPredicate = datePredicate;
+            }
+        }
+        if (combinedPredicate == null) {
+            combinedPredicate = food -> true;
+        }
```

Comment:
Instead of using null, how about declaring a subclass of FoodContainPredicate that always returns true, then use
`FoodContainsKeywordsPredicate foodPredicate = foodPredicateParameter.getValue().orElseGet(THAT_CLASS_NAME::new]);`.

Reply:
Yep, was thinking of doing this. I used null just to make the code functional, will refactor tomorrow probably.

Reply:
Instead of a dummy class, do you think doing `foodPredicateParameter.getValue().orElse(food -> true);`would be better?

Reply:
yea this works too